lib/drivers/insuletDriver.js

```diff
@@ -206,7 +206,24 @@ module.exports = function (config) {
       'immediate_duration_seconds'
     ], postprocess: function (rec) {
       rec.volume_units = toUnits(rec.volume);
-      rec.extended_duration_msec = rec.extended_duration_minutes * sundial.MIN_TO_MSEC;
+      // this occurs if an extended bolus is programmed but the bolus is interrupted
+      // before any of the extended portion is delivered!
+      if (rec.extended_duration_minutes === 65535) {
```

Comment:
That's a good question. Do you mean guarantee that the *only* situation that produces this is an interrupted bolus? I don't know.

Or do you mean guarantee that an interrupted bolus (with no extended delivered, as specified) produces the special value 65535? Yes, I believe we can be confident in that. Quoting from @kentquirk "Look at the number -- it's 65535 -- that's 0xFFFF -- or -1 if you treat it as a 16-bit signed number." It's clearly a special value, not just a weird constant.